scripts/modules_auto_pr/modules_auto_pr.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+set -e  # Exit immediately if a command exits with a non-zero status
+set -x  # Print commands and their arguments as they are executed
+set -u  # Treat unset variables as an error when substituting
+
+####################################################################################################
+#                                            VARIABLES                                             #
+####################################################################################################
+
+TIMESTAMP=$(date +"%Y-%m-%d_%H-%M")
+HUMAN_TIMESTAMP=$(date "+%-d %B %Y, %H:%M")
+REPO_URL="git@github.com:hpcugent/vsc_user_docs" # The repository for which the script will generate the PR
+BASE_BRANCH="main"
+BRANCH_NAME="auto_update_modules_$TIMESTAMP"
+REPO_NAME="vsc_user_docs" # name of the cloned repo. script available_modules.py requires this to be the name. Do not change.
+REPO_PATH="/tmp/$USER/modules_auto_pr_script_$TIMESTAMP/$REPO_NAME"
+COMMIT_MESSAGE="Update Modules [$HUMAN_TIMESTAMP]"
+PR_TITLE="Auto Update Modules [$HUMAN_TIMESTAMP]"
+
+make_pr_body() {
+  local n_added_modules="$1"
+  local n_removed_modules="$2"
+
+  echo "This is an automated pull request updating the markdown files of all available modules."
+  echo ""
+  echo "Changes:"
+  echo "- Updated the markdown files of all previously available modules"
+  echo "- Added $n_added_modules new modules"
+  echo "- Removed $n_removed_modules modules"
+}
+
+####################################################################################################
+#                                             SCRIPT                                               #
+####################################################################################################
+
+echo_info()  { echo -e "\e[32m$0: [INFO] $1\e[0m"; }
+echo_error() { echo -e "\e[31m$0: [ERROR] $1\e[0m"; }
+
+main() {
+  local github_pat_file="$1"
+  local fork_user="$2" # The user that owns the fork from which the script will create the PR. e.g. lbarraga
+  local fork_url="git@github.com:$fork_user/vsc_user_docs" # The fork from which the script will create the PR
+
+  # Check if the `gh` command is available
+  if ! command -v gh &> /dev/null; then
+      echo_error "The GitHub CLI (gh) is not installed."
+      echo_error "Please install it by following the instructions at https://cli.github.com."
+      exit 1
+  fi
+
+  # Check if PAT file is present and fork user is provided in one if statement
+  if [ ! -f "$github_pat_file" ] || [ -z "$fork_user" ]; then
```

Comment:
If `PAT` is an empty file, this gets stuck with:

```
+ gh auth login --with-token

! First copy your one-time code: 0BBF-EBC9
Open this URL to continue in your web browser: https://github.com/login/device
```

Wrong content is fine:
```
error validating token: HTTP 401: Bad credentials (https://api.github.com/)
Try authenticating with:  gh auth login
```

Easy fix is to use `-f` rather than `-s` here (TIL):

```suggestion
  # -s checks whether file exists and has non-zero size (is not empty)
  if [ ! -s "$github_pat_file" ] || [ -z "$fork_user" ]; then